### ISSUE TITLE
Really weird issue when svgo is executed in a subshell from Python

### ISSUE THREAD

**Issue by tssajo** [svg/svgo#345]
I'm on Windows 7, Node.js v0.10.38, svgo v0.5.1, I installed `svgo` globally with: `npm install -g svgo`

Here is my svgo command:

`svgo C:\_projects\_test\car.svg C:\_projects\_test\car.min.svg`

Here is what I experience:

- When I execute the above command via Windows command prompt (CMD.exe) it works fine.

- When I execute it from my Python script like this:<br/>
subprocess.call(['svgo', 'C:\\_projects\\_test\\car.svg', 'C:\\_projects\\_test\\car.min.svg'], shell=True)<br/>
then it still works fine.

- But I want to capture the STDERR output (if any) so I try to execute it like this from my Python script:<br/>
p = subprocess.Popen(['svgo', 'C:\\_projects\\_test\\car.svg', 'C:\\_projects\\_test\\car.min.svg'], stderr=subprocess.PIPE, shell=True)
err = p.communicate()[1]
result = p.returncode<br/>
then it does not work! svgo starts to run that's for sure but the created output file is 0 bytes long!

I have no idea why `svgo` fails this way (creates a 0 bytes long output file) when its STDERR output is redirected (PIPE'd) to my Python program. I do run other Node.js CLI programs the very same way such as `uglifyjs`, `cleancss`, `html-minifier` but those other programs work fine even when I launch them from Python via the subprocess.Popen method. Only `svgo` fails this way.

Note: The same issue is present with svgo 0.5.0, I did not test it with older versions.

**Comment by GreLI** [comment]
I have no idea, svgo uses standard ``console.error()`` method, so if it doens't go to STDERR, I guess it should be addressed to node.js.

**Comment by tssajo** [comment]
Yeah, but the main problem is that the resulting minified(!) file is 0 bytes long. (Not talking about the STDERR output here.)

So I'm guessing here, maybe because svgo somehow detects that STDERR is being PIPEd, it is maybe trying to read the input (original svg data) from STDIN too?? This would explain the 0 bytes long minified file output.

Again, the other minifier CLI tools (e.g. `uglifyijs`, etc.) are also written in node.js and they work just fine when I run them from Python's subprocess.Popen method. Only `svgo` fails, it does not minify the specified input file, it creates a 0 bytes long minified file instead. Which I would say is a major problem.

**Comment by GreLI** [comment]
> So I'm guessing here, maybe svgo somehow detects that STDERR is being PIPEd then in turn it may try to get the input (original svg data) from STDIN as well?

No, svgo does nothing of that.

Btw, have you tried node 0.12? Or io.js. node.js may have some issues.

**Comment by tssajo** [comment]
Not, it is not realated to node version.

I've made really good progress on this issue, here are my findings:

By entering the following command into a Windows command prompt (CMD.exe) :

`svgo car.svg car.min.svg 2 > null`

I get the zero bytes long minified file as a result and I can see an error message on the screen which is coming from `svgo` I assume:

`Unknown argument: 2`

But if I run this:

`svgo car.svg car.min.svg 2> null`

Notice the difference: there is no space between `2` and `>`.

So maybe on Windows, when `shell=True` provided for Python's pOpen function then Python places a space there between the `2` and the `>` characters? I'm not sure about this, but this would be a reasonable explanation.

Another thing which I found out is this:
If I do redirect both STDOUT and STDERR like this:

`p = subprocess.Popen(['svgo', 'C:\_projects\_test\car.svg', 'C:\_projects\_test\car.min.svg'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)`

then I do not encounter the problem!
I do not exactly know why, but at least it gives me a workaround solution.
